app/containers/MessageBox/RecordAudio.js

```diff
@@ -0,0 +1,215 @@
+import { getInfoAsync } from 'expo-file-system';
+import { deactivateKeepAwake, activateKeepAwake } from 'expo-keep-awake';
+
+import styles from './styles';
+import I18n from '../../i18n';
+import { themes } from '../../constants/colors';
+import { CustomIcon } from '../../lib/Icons';
+
+const RECORDING_EXTENSION = '.aac';
+const RECORDING_SETTINGS = {
+	android: {
+		extension: RECORDING_EXTENSION,
+		outputFormat: Audio.RECORDING_OPTION_ANDROID_OUTPUT_FORMAT_AAC_ADTS,
+		audioEncoder: Audio.RECORDING_OPTION_ANDROID_AUDIO_ENCODER_AAC,
+		sampleRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.android.sampleRate,
+		numberOfChannels: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.android.numberOfChannels,
+		bitRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.android.bitRate
+	},
+	ios: {
+		extension: RECORDING_EXTENSION,
+		audioQuality: Audio.RECORDING_OPTION_IOS_AUDIO_QUALITY_MIN,
+		sampleRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.sampleRate,
+		numberOfChannels: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.numberOfChannels,
+		bitRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.bitRate,
+		outputFormat: Audio.RECORDING_OPTION_IOS_OUTPUT_FORMAT_MPEG4AAC
+	}
+};
+const RECORDING_MODE = {
+	allowsRecordingIOS: true,
+	playsInSilentModeIOS: true,
+	staysActiveInBackground: false,
+	shouldDuckAndroid: true,
+	playThroughEarpieceAndroid: false,
+	interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
+	interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX
+};
+
+const formatTime = function(seconds) {
+	let minutes = Math.floor(seconds / 60);
+	seconds %= 60;
+	if (minutes < 10) { minutes = `0${ minutes }`; }
+	if (seconds < 10) { seconds = `0${ seconds }`; }
+	return `${ minutes }:${ seconds }`;
+};
+
+export default class RecordAudio extends React.PureComponent {
+	static propTypes = {
+		theme: PropTypes.string,
+		recordingCallback: PropTypes.func,
+		onFinish: PropTypes.func
+	}
+
+	constructor(props) {
+		super(props);
+		this.isRecorderBusy = false;
+		this.state = {
+			isRecording: false,
+			recordingDurationMillis: 0
+		};
+	}
+
+	componentDidUpdate() {
+		const { recordingCallback } = this.props;
+		const { isRecording } = this.state;
+
+		recordingCallback(isRecording);
+	}
+
+	componentWillUnmount() {
+		if (this.recording) {
+			this.cancelRecordingAudio();
+		}
+	}
+
+	get duration() {
+		const { recordingDurationMillis } = this.state;
+		return formatTime(Math.floor(recordingDurationMillis / 1000));
+	}
+
+	isRecordingPermissionGranted = async() => {
+		const permission = await Audio.getPermissionsAsync();
+		return permission.status === 'granted';
+	}
+
+	onRecordingStatusUpdate = (status) => {
+		this.setState({
+			isRecording: status.isRecording,
+			recordingDurationMillis: status.durationMillis
+		});
+	}
+
+	startRecordingAudio = async() => {
+		if (!this.isRecorderBusy) {
+			this.isRecorderBusy = true;
+			try {
+				if (await this.isRecordingPermissionGranted()) {
```

Comment:
no use `await` inner a conditional.
My main idea was do something like that:
```javascript
isRecordingPermissionGranted = () => {
  try {
    const permission = await Audio.getPermissionsAsync();
    if (permission.status === 'granted') {
      return true;
    }
    await Audio.requestPermissionsAsync();
  } catch {
    // Do nothing
  }
  return false;
}
```
and here:
```javascript
const canRecord = await this.isRecordingPermissionGranted();
if (canRecord) {
...
}
```